evaluation/few_shot_eval.py

```diff
@@ -0,0 +1,699 @@
+        format="%(asctime)s - %(levelname)s - %(name)s - %(message)s",
+        datefmt="%m/%d/%Y %H:%M:%S",
+        level=logging.INFO,
+    )
+    logger.info(accelerator.state)
+
+    # Setup logging, we only want one process per machine to log things on the screen.
+    # accelerator.is_local_main_process is only True for one process per machine.
+    logger.setLevel(logging.INFO if accelerator.is_local_main_process else logging.ERROR)
+    if accelerator.is_local_main_process:
+        datasets.utils.logging.set_verbosity_warning()
+        transformers.utils.logging.set_verbosity_info()
+    else:
+        datasets.utils.logging.set_verbosity_error()
+        transformers.utils.logging.set_verbosity_error()
+
+
+    # Handle the output directory creation
+    if accelerator.is_main_process:
+        os.makedirs(args.output_dir, exist_ok=True)
+    accelerator.wait_for_everyone()
+
+    # In distributed evaluation, the load_dataset function guarantee that only one local process can concurrently
+    # download the dataset.
+    if args.dataset_name is not None:
+        # Downloading and loading a dataset from the hub.
+        if args.dataset_name == "anli":
+            raw_train_dataset = load_dataset(args.dataset_name, split=f'train_{args.dataset_config_name}')  # dataset_config_name = "r1", "r2", or "r3"
+            raw_eval_dataset = load_dataset(args.dataset_name, split=f'dev_{args.dataset_config_name}')
+        else:
+            raw_train_dataset = load_dataset(args.dataset_name, args.dataset_config_name, split="train")
+            raw_eval_dataset = load_dataset(args.dataset_name, args.dataset_config_name, split="validation")
+    else:
+        raise ValueError('Please specify `args.dataset_name` and `args.dataset_config_name` as appear in `promptsource`.')
+    #TODO(Victor): enable loading pre-processed dataset from https://huggingface.co/datasets/bigscience/P3
+
+    # Trim a number of evaluation examples
+    if args.debug:
+        raw_train_dataset = raw_train_dataset.select(range(100))
+        raw_eval_dataset = raw_eval_dataset.select(range(100))
+
+    column_names = raw_eval_dataset.column_names
+
+
+    # Load pretrained model and tokenizer
+    #
+    # In distributed training, the .from_pretrained methods guarantee that only one local process can concurrently
+    # download model & vocab.
+    if args.config_name:
+        config = AutoConfig.from_pretrained(args.model_name_or_path)
+    elif args.model_name_or_path:
+        config = AutoConfig.from_pretrained(args.model_name_or_path)
+    else:
+        raise ValueError(
+            "Either `args.config_name` or `args.model_name_or_path` should be provided."
+        )
+
+    if args.tokenizer_name:
+        tokenizer = AutoTokenizer.from_pretrained(args.tokenizer_name, use_fast=not args.use_slow_tokenizer)
+    elif args.model_name_or_path:
+        tokenizer = AutoTokenizer.from_pretrained(args.model_name_or_path, use_fast=not args.use_slow_tokenizer)
```

Comment:
I had this error, because of missing protobuff.

```
ImportError:
T5Converter requires the protobuf library but it was not found in your environment. Checkout the instructions on the
installation page of its repo: https://github.com/protocolbuffers/protobuf/tree/master/python#installation and follow the ones
that match your environment.
```

If you follow the provided link, it does not really contain installation instructions 😢. All they have is "Build the C++ code, or install a binary distribution of protoc." without explanation on how to install protoc. I found an easier way to do it (requires Ubuntu>=20.04).

```
sudo apt-get install protobuf-compiler
pip install protobuff
```

Recommend adding this to README.

Reply:
`pip install protobuf` usually fixed the issue for me
+1 documentation